c++/src/kj/refcount.h

```diff
@@ -143,7 +144,8 @@ Own<T> Refcounted::addRefInternal(T* object) {
 }
 
 template <typename T>
-Rc<T> Refcounted::addRcRefInternal(T* object) {
+Rc<T> Refcounted::addPtrRefInternal(T* object) {
+  static_assert(std::is_base_of_v<Refcounted, T>);
```

Comment:
You can use `kj::canConvert<T&, Refcounted&>()` to avoid bringing in `<type_traits>` here.